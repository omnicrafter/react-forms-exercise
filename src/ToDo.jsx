const ToDo = ({ removeToDo, task, toDoKey }) => {
  return (
    <>
      <li>{task}</li>
      <button onClick={() => removeToDo(toDoKey)}>remove</button>
    </>
  );
};

export default ToDo;
