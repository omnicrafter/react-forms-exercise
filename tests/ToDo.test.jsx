import { describe, it, expect } from "vitest";
import { render, screen, fireEvent } from "@testing-library/react";
import ToDo from "../src/ToDo";

describe("ToDo", () => {
  it("renders without crashing", () => {
    render(<ToDo />);
  });

  it("matches snapshot", () => {
    const { asFragment } = render(
      <ToDo toDoKey="1" task="test task" removeToDo={() => {}} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("calls the removeToDo function when the button is clicked", () => {
    const removeToDo = vitest.fn();
    const { getByText } = render(
      <ToDo toDoKey="1" task="test task" removeToDo={removeToDo} />
    );

    fireEvent.click(getByText("remove"));

    expect(removeToDo).toHaveBeenCalledTimes(1);
    expect(removeToDo).toHaveBeenCalledWith("1");
  });
});
