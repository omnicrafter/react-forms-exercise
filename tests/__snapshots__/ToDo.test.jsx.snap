// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ToDo > matches snapshot 1`] = `
<DocumentFragment>
  <li>
    test task
  </li>
  <button>
    remove
  </button>
</DocumentFragment>
`;
